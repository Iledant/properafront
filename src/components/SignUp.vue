<template>
  <v-dialog :value="value" @input="$emit('input', false)" max-width="600px" persistent>
    <v-card>
      <v-card-title class="primary white--text title">Inscription à Propera</v-card-title>
      <v-container grid-list-md fluid>
        <v-layout wrap>
          <v-text-field
            label="Nom complet"
            v-model="name"
            :rules="[checkIfNotEmpty]"
            required
            autofocus
            data-cy="signUpFullName"
          />
        </v-layout>
        <v-layout row>
          <v-text-field
            label="email"
            v-model="email"
            :rules="[checkIfNotEmpty, emailCheck]"
            required
            data-cy="signUpEmail"
          />
        </v-layout>
        <v-layout row>
          <v-text-field
            label="Mot de passe"
            v-model="password"
            type="password"
            :rules="[checkIfNotEmpty]"
            @keyup.enter="onCreate"
            required
            data-cy="signUpPassword"
          />
        </v-layout>
      </v-container>
      <v-card-actions class="tertiary">
        <v-spacer />
        <v-btn text color="primary" @click="$emit('input', false)" data-cy="signUpCancel">Annuler</v-btn>
        <v-btn
          text
          color="primary"
          @click="onCreate"
          :disabled="disabled"
          :loading="loading"
          data-cy="signUpOk"
        >S'inscrire</v-btn>
      </v-card-actions>
    </v-card>
  </v-dialog>
</template>

<script>
import checkIfNotEmpty from './Mixins/CheckIfNotEmpty'
import * as types from '../store/mutation-types.js'
export default {
  name: 'SignUp',
  mixins: [checkIfNotEmpty],
  props: {
    value: { type: Boolean, default: false }
  },
  data: () => ({
    name: '',
    password: '',
    email: ''
  }),
  computed: {
    disabled () {
      return this.name === '' || this.password === '' || this.email === ''
    },
    loading () {
      return this.$store.getters.loading
    }
  },
  methods: {
    emailCheck (input) {
      /* eslint-disable no-useless-escape */
      const re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
      /* eslint-enable no-useless-escape */
      return re.test(input) ? true : 'Adresse email attendue'
    },
    async onCreate () {
      if (!this.disabled) {
        try {
          await this.$http.post('user/signup', {
            name: this.name,
            email: this.email,
            password: this.password
          })
          this.$emit('success')
          this.$emit('input', false)
        } catch (response) {
          if (response.status === 422) {
            this.$store.commit(types.SET_ERROR_MESSAGE,
              'L\'adresse email est déjà utilisée')
          }
        }
      }
    }
  }
}
</script>
