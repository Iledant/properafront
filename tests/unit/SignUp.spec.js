import { mount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import Vuetify from 'vuetify'
import SignUp from '@/components/SignUp.vue'

const localVue = createLocalVue()

describe('SignUp', () => {
  let vuetify
  let store

  beforeEach(() => {
    vuetify = new Vuetify()
    store = new Vuex.Store({ getters: { loading: () => true } })
  })

  it('should not render if value isn\'t passed', () => {
    const wrapper = mount(SignUp, {
      localVue,
      vuetify,
      store,
      propsData: { value: false }
    })
    expect(wrapper).toMatchSnapshot()
  })

  it('should not render if value is true', () => {
    const wrapper = mount(SignUp, {
      localVue,
      vuetify,
      store,
      propsData: { value: true }
    })
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.props('value')).toBe(true)
  })
})
