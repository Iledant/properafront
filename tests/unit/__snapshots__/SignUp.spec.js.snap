// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp should not render if value is true 1`] = `
<div role="dialog" class="v-dialog__container">
  <div role="document" tabindex="0" class="v-dialog__content v-dialog__content--active" style="z-index: 202;">
    <transition-stub name="dialog-transition" appear="true">
      <div class="v-dialog v-dialog--active v-dialog--persistent" style="transform-origin: center center; max-width: 600px;">
        <div class="v-card v-sheet theme--light">
          <div class="v-card__title primary white--text">Inscription Ã  Propera</div>
          <div class="container container--fluid grid-list-md">
            <div class="layout wrap">
              <div class="v-input theme--light v-text-field">
                <div class="v-input__control">
                  <div class="v-input__slot">
                    <div class="v-text-field__slot"><label for="input-15" class="v-label theme--light" style="left: 0px; position: absolute;">Nom complet</label><input required="required" data-cy="signUpFullName" autofocus="autofocus" id="input-15" type="text"></div>
                  </div>
                  <div class="v-text-field__details">
                    <div class="v-messages theme--light">
                      <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div class="layout row">
              <div class="v-input theme--light v-text-field">
                <div class="v-input__control">
                  <div class="v-input__slot">
                    <div class="v-text-field__slot"><label for="input-18" class="v-label theme--light" style="left: 0px; position: absolute;">email</label><input required="required" data-cy="signUpEmail" id="input-18" type="text"></div>
                  </div>
                  <div class="v-text-field__details">
                    <div class="v-messages theme--light">
                      <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div class="layout row">
              <div class="v-input theme--light v-text-field">
                <div class="v-input__control">
                  <div class="v-input__slot">
                    <div class="v-text-field__slot"><label for="input-21" class="v-label theme--light" style="left: 0px; position: absolute;">Mot de passe</label><input required="required" data-cy="signUpPassword" id="input-21" type="password"></div>
                  </div>
                  <div class="v-text-field__details">
                    <div class="v-messages theme--light">
                      <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div class="v-card__actions tertiary">
            <div class="spacer"></div> <button type="button" class="v-btn v-btn--flat v-btn--text theme--light v-size--default primary--text" data-cy="signUpCancel"><span class="v-btn__content">
        Annuler
      </span></button> <button type="button" disabled="disabled" class="v-btn v-btn--disabled v-btn--flat v-btn--loading v-btn--text theme--light v-size--default" data-cy="signUpOk"><span class="v-btn__content">S'inscrire</span><span class="v-btn__loader"><div role="progressbar" aria-valuemin="0" aria-valuemax="100" class="v-progress-circular v-progress-circular--indeterminate" style="height: 23px; width: 23px;"><svg xmlns="http://www.w3.org/2000/svg" viewBox="21.904761904761905 21.904761904761905 43.80952380952381 43.80952380952381" style="transform: rotate(0deg);"><circle fill="transparent" cx="43.80952380952381" cy="43.80952380952381" r="20" stroke-width="3.8095238095238093" stroke-dasharray="125.664" stroke-dashoffset="125.66370614359172px" class="v-progress-circular__overlay"></circle></svg><div class="v-progress-circular__info"></div></div></span></button>
          </div>
        </div>
      </div>
    </transition-stub>
  </div>
</div>
`;

exports[`SignUp should not render if value isn't passed 1`] = `
<div role="dialog" class="v-dialog__container">
  <!---->
</div>
`;
